dartagnan/src/main/java/com/dat3m/dartagnan/wmm/analysis/RelationAnalysis.java

```diff
@@ -47,20 +53,19 @@
 import static com.dat3m.dartagnan.wmm.relation.RelationNameRepository.CO;
 import static com.dat3m.dartagnan.wmm.relation.RelationNameRepository.RF;
 import static com.google.common.base.Preconditions.checkNotNull;
+import static com.google.common.base.Preconditions.checkState;
 import static com.google.common.base.Verify.verify;
 import static com.google.common.collect.Lists.reverse;
 import static com.google.common.collect.Sets.difference;
 import static com.google.common.collect.Sets.intersection;
 import static java.util.stream.Collectors.toList;
-import static java.util.stream.Collectors.toSet;
-import static java.util.stream.IntStream.iterate;
 
 @Options
 public class RelationAnalysis {
 
     private static final Logger logger = LogManager.getLogger(RelationAnalysis.class);
 
-    private static final Set<Tuple> EMPTY_SET = new HashSet<>();
+    private static final HashSet<Tuple> EMPTY_SET = new HashSet<>();
```

Comment:
You should keep using `Set<Tuple>` unless the more concrete type exposes more functionality (which it does not IIRC). The JIT compiler is aware of the true type, so you will not get any performance benefit whatsoever when using a concrete type.